src/core/Akka/Actor/ActorBase.cs

```diff
@@ -155,6 +159,38 @@ protected static IActorContext Context
         /// <returns>TBD</returns>
         protected internal virtual bool AroundReceive(Receive receive, object message)
         {
+            if (message is Scheduler.TimerScheduler.ITimerMsg tm)
+            {
+                if (this is IWithTimers withTimers && withTimers.Timers is Scheduler.TimerScheduler timers)
```

Comment:
now with the `IWithTimers` we end up here with two more type checks, but only for timer messages